server.js

```diff
@@ -0,0 +1,247 @@
+const http = require('http');
+const fs = require('fs');
+const qs = require('querystring');
+
+const authorizedUsers = {
+  banana: 'peel',
+};
+
+const server = http.createServer(function(req, res) {
+  // GET
+
+  if (req.method === 'GET') {
+    if (req.url === '/') {
+      fs.readFile('./public/index.html', 'utf8', (err, data) => {
+        if (err) {
+          return res.end(`Error: ${err}`);
+        } else {
+          return res.end(data);
+        }
+      });
+    }
+
+    fs.readFile(`./public${req.url}`, 'utf8', (err, data) => {
+      if (err) {
+        return res.end(`Error: ${err}`);
+      } else {
+        return res.end(data);
+      }
+    });
+  }
+
+  // POST
+
+  if (req.method === 'POST' && req.url === '/elements') {
+    if (req.headers.authorization) {
+      const encodedString = req.headers.authorization.slice(6);
+      const base64Buffer = new Buffer(encodedString, 'base64');
+      const decodedString = base64Buffer.toString();
+      let username = decodedString.slice(0, decodedString.indexOf(':'));
+      let password = decodedString.slice(decodedString.indexOf(':') + 1);
+
+      if (authorizedUsers.hasOwnProperty(username) && authorizedUsers[username] === password) {
+        let body = '';
+
+        req.on('data', (chunk) => {
+          body += chunk;
+        });
+
+        req.on('end', (chunk) => {
+          if (chunk) {
+            body += chunk;
+          } else {
+            let parsedBody = qs.parse(body);
+            let pageTemplate = `<!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+            <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+            <title>Document</title>
+            <link rel="stylesheet" href="css/styles.css" />
+          </head>
+          <body>
+          <h1>${parsedBody.elementName}</h1>
+          <h2>${parsedBody.elementSymbol}</h2>
+          <h3>${parsedBody.elementAtomicNumber}</h3>
+          <p>${parsedBody.elementDescription}</p>
+          <p><a href="/">back</a></p>
+          </body>
+        </html>`;
+            let elementName = parsedBody.elementName.toLowerCase();
+            fs.writeFile(`./public/${elementName}.html`, pageTemplate, (err, data) => {
+              if (err) {
+                res.writeHead(500);
+                return res.end('{ "success": false }');
+              } else {
+                updateIndex();
```

Comment:
updateIndex should take in a function parameter that you define inline here. Then you can put the res.writeHead and the res.end in that inline function to be called after the index is updated.